### ISSUE TITLE
Calculating loss during testing

### ISSUE THREAD

**Issue by micophilip** [zjunlp/DeepKE#96]
**Describe the question**
Hello, I noticed in [train.py](https://github.com/zjunlp/DeepKE/blob/main/src/deepke/name_entity_re/few_shot/module/train.py#L169-L177), loss is only calculated during training. I'd like to calculate loss during prediction as well but looks like the tensor coming back from `self.model.predict` is not a logits tensor like the one coming from `self.model()`

Any suggestion on how I can compute the loss during `test` mode is appreciated.

```
elif mode == "test":  # test: just get pred
    src_tokens, src_seq_len, first = batch
    pred = self.model.predict(src_tokens, src_seq_len, first)
    return pred
else:  # train: get loss
    src_tokens, tgt_tokens, src_seq_len, tgt_seq_len, first, target_span = batch
    pred = self.model(src_tokens, tgt_tokens, src_seq_len, first)
    loss = self.loss(tgt_tokens, tgt_seq_len, pred)
    return loss
```

**Environment (please complete the following information):**
 - OS: mac
 - Python Version 3.8

**Comment by Flow3rDown** [comment]
Hi,
Since the test set does not always contain labels, we did not calculate the loss in the testing phase. Maybe you can rewrite the evalua code to get the loss.
